samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -316,6 +345,19 @@ private void onShutDown() {
     }
   }
 
+  private void shutDownContainerPlacementHandlerAndUtils() {
+    // Shutdown container placement handler
+    containerPlacementHandler.stop();
+    try {
+      containerPlacementHandlerThread.join();
+      LOG.info("Stopped container placement handler thread");
+      containerPlacementUtil.stop();
+    } catch (InterruptedException ie) {
+      LOG.error("Container Placement handler thread join threw an interrupted exception", ie);
```

Comment:
This error log assumes `containerPlacementUtil.stop()` will never throw an exception. Is that true?

Reply:
+1, perhaps use finally to do a clean guaranteed cleanup?